Security/TwoFactor/Provider/U2F/TwoFactorProvider.php

```diff
@@ -46,23 +38,20 @@ public function __construct(U2FAuthenticatorInterface $authenticator, TwoFactorF
         $this->session = $session;
     }
 
-    /**
-     * beginAuthentication
-     * @param AuthenticationContextInterface $context
-     * @return boolean
-     **/
+    public function prepareAuthentication($user): void
```

Comment:
We should be consistent with the return types. Some are defined in the docblock and some are real return types.

Reply:
In this case we're bound to `\Scheb\TwoFactorBundle\Security\TwoFactor\Provider\TwoFactorProviderInterface`. 

In my opinion we should use return types where possible and use docblocks only where return types aren't possible (because of BC or other dependencies). But this PR doesn't cover the whole project, I mainly focused on v4 compatibility.